css/DateInput.scss

```diff
@@ -50,6 +50,13 @@ $caret-top: $react-dates-spacing-vertical-picker - $react-dates-width-tooltip-ar
   border: 0;
   height: 100%;
   width: 100%;
+
+  &[readonly] {
+    user-select: none;
```

Comment:
the standard property should always go last, so that it takes precedence in browsers that support both.